pgxn/neon/walproposer.c

```diff
@@ -119,6 +119,10 @@ WalProposerCreate(WalProposerConfig *config, walproposer_api api)
 		{
 			wp_log(FATAL, "failed to parse neon.safekeepers generation number: %m");
 		}
+		if (*endptr != ':')
```

Comment:
nit: Maybe reuse `split_off_safekeepers_generation` here? Looks exactly like it

Reply:
done